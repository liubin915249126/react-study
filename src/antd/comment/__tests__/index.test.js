import React from 'react';
import { mount } from 'enzyme';
import Comment from '../index';
import mountTest from '../../../tests/shared/mountTest';
import rtlTest from '../../../tests/shared/rtlTest';

describe('Comment', () => {
  mountTest(Comment);
  rtlTest(Comment);

  it('should support empty actions', () => {
    const wrapper = mount(
      <Comment
        actions={[]}
        author={<a>Han Solo</a>}
        content={
          <p>
            We supply a series of design principles, practical patterns and high quality design
            resources (Sketch and Axure), to help people create their product prototypes beautifully
            and efficiently.
          </p>
        }
        datetime="YYYY-MM-DD HH:mm:ss"
      />,
    );
    expect(wrapper).toMatchRenderedSnapshot();
  });
});
