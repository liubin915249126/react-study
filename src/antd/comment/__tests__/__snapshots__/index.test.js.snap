// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comment rtl render component should be rendered correctly in RTL direction 1`] = `
<div
  class="ant-comment ant-comment-rtl"
>
  <div
    class="ant-comment-inner"
  >
    <div
      class="ant-comment-content"
    >
      <div
        class="ant-comment-content-detail"
      />
    </div>
  </div>
</div>
`;

exports[`Comment should support empty actions 1`] = `
<div
  class="ant-comment"
>
  <div
    class="ant-comment-inner"
  >
    <div
      class="ant-comment-content"
    >
      <div
        class="ant-comment-content-author"
      >
        <span
          class="ant-comment-content-author-name"
        >
          <a>
            Han Solo
          </a>
        </span>
        <span
          class="ant-comment-content-author-time"
        >
          YYYY-MM-DD HH:mm:ss
        </span>
      </div>
      <div
        class="ant-comment-content-detail"
      >
        <p>
          We supply a series of design principles, practical patterns and high quality design resources (Sketch and Axure), to help people create their product prototypes beautifully and efficiently.
        </p>
      </div>
    </div>
  </div>
</div>
`;
